week5/palindromeLL.cc

```diff
@@ -0,0 +1,176 @@
+/* palindromeLL.cc
+ * author: connie
+ *
+ * LINKED LIST PALINDROME
+ * checks if a LL is a palindrome.
+ */
+
+#include <cstdlib>
+#include <iostream>
+#include <cstdio>
+#include <string>
+#include <stack>
+#include <cmath>
+#include <cassert>
+
+using namespace std;
+
+//== DEFINITIONS
+struct node
+{
+  char data;
+  node *next;
+  node(char d) : data(d), next(NULL) {}
+};
+
+//== DECLARATIONS
+void insert(node*& head, int key);
+void deleteLL(node*& head);
+bool palindromeLL(node *head);
+
+//== FUNCTIONS
+/*
+ * void insert(node*& head, int key)
+ * inputs: head - the LL node to insert the new node in front of
+           key - value of the new node to be inserted
+ * output: none
+ * notes: helper function to insert a new node into the LL, as the new head node
+ */
+void insert(node*& head, char d)
+{
+  node *curr = new node(d);
+  curr->next = head;
+
+  head = curr;
+}
+
+/*
+ * void deleteLL(node*& head)
+ * inputs: head - the start of the entire LL that we want to delete
+ * output: none
+ * notes: helper function to prevent memory leaks
+ */
+void deleteLL(node*& head)
+{
+  node *del = head;
+
+  while (del) {
+    head = head->next;
+    delete del;
+    del = head;
+  }
+
+  head = NULL;
+}
+
+
+/*
+ * bool palindromeLL(node *head)
+ * inputs: *head - pointer to the starting node of the LL
+ * output: T if the LL is a palindrome, F if not
+ * notes: main solution function
+ */
+bool palindromeLL(node *head)
+{
+  // case: null or only one -> is a palindrome
+  if (!head || (!head->next)) return true;
```

Comment:
Just a little detail here, I think that you should return an exception/error when the head is null

Reply:
Definitely something to ask the interviewer about. I saw it in your Talk TEBOW IT notes so I know it was something you considered

Reply:
fixed to throw null exception